NeuroID/src/main/java/com/neuroid/tracker/service/NIDJobServiceManager.kt

```diff
@@ -48,61 +51,112 @@ class NIDJobServiceManager(
         this.userActive = true
         this.clientKey = clientKey
         this.application = application
-        jobCaptureEvents = createJobServer()
+
+        sendEventsJob = createSendEventsServer()
+        sendCadenceJob = createSendCadenceServer()
         NIDSensorHelper.initSensorHelper(application, logger)
 
-        gyroCadenceJob = createGyroJobServer()
+        gyroCadenceJob = createGyroCadenceServer()
         activityManager = application.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
         this.isSetup = true
     }
 
     @Synchronized
     fun stopJob() {
         NIDSensorHelper.stopSensors()
-        jobCaptureEvents?.cancel()
-        jobCaptureEvents = null
+        sendCadenceJob?.cancel()
+        sendCadenceJob = null
 
         gyroCadenceJob?.cancel()
         gyroCadenceJob = null
     }
 
     @Synchronized
     fun isStopped(): Boolean {
-        return jobCaptureEvents?.isActive != true
+        return sendCadenceJob?.isActive != true
     }
 
     @Synchronized
     fun restart() {
         this.userActive = true
         NIDSensorHelper.restartSensors()
-        jobCaptureEvents?.cancel()
-        jobCaptureEvents = createJobServer()
+
+        // If the job has ended for any reason restart it
+        if (!sendEventsJob.isActive) {
+            sendEventsJob.cancel()
+            sendEventsJob = createSendEventsServer()
+        }
+
+        sendCadenceJob?.cancel()
+        sendCadenceJob = createSendCadenceServer()
 
         gyroCadenceJob?.cancel()
-        gyroCadenceJob = createGyroJobServer()
+        gyroCadenceJob = createGyroCadenceServer()
+    }
+
+    /**
+     * Notify the send events server to initiate a send.
+     */
+    fun sendEvents(forceSendEvents: Boolean = false) {
+        CoroutineScope(dispatcher).launch {
+            sendEventsNotification.send(forceSendEvents)
+        }
```

Comment:
Since `sendEventNotification` has an unlimited buffer, I think we can drop the coroutine and use `trySend`, from [Channel](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-channel/):

`[Sending](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-channel/send.html) to this channel never suspends, and [trySend](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-channel/try-send.html) always succeeds.`